ert/data/record/_transmitter.py

```diff
@@ -49,6 +51,22 @@ class RecordTransmitterType(Enum):
     shared_disk = auto()
 
 
+def _unflatten_record_dict(
+    flat_record_dict: Union[Dict[str, BlobRecord], Dict[str, NumericalRecord]]
+) -> Dict[str, Any]:
+    record_dict: Dict[str, Any] = {}
+    for rec_path, rec in flat_record_dict.items():
+        keys = rec_path.split("/")
+        sub_rec_name = keys[-1]
+        branch = record_dict
+        for key in keys[:-1]:
+            if key not in branch:
+                branch[key] = {}
+            branch = branch[key]
+        branch[sub_rec_name] = rec
+    return record_dict
```

Comment:
I think this function could benefit from some renaming of variable names. At least `branch` seems to be more like the current node in a tree traversal.

Reply:
I think `branch` is okay and meaningful, considering it never becomes a leaf node? 

I initially thought to mention that there seems to some nonzero distance to (reconstruction through) traversal algorithms, since we are not using the ordering-information in the dict (it is even mutable) to reconstruct the tree. But the keys are immutable, and contains the tree-structure information (not deconstructed), so is it fair to say that we traversing a tree here? Just curious. :)

Reply:
It constructs a tree from multiple paths from root to leaf. Guess the branch name is ok, but imo it represents the current sub-tree in the traversal. But this is all isolated inside a 10-line private function, so I'm fine leaving it as is :slightly_smiling_face:

Reply:
:relieved:

Reply:
I concur with the ☺️ emoji, and also everything else. If the internal nodes were of the `Record` type I too would like it to be called current node, I think. It is a cool 10-line private function though (trees and traversal usually are) 🌴.

Kinda relevant to this: Is deconstruction through traversal algorithms (perhaps multiple are needed) using the ordering of the flattened (also internal) nodes still possible considering this is not a binary tree? I have only seen the latter problem before.

Reply:
I don't know why it shouldn't work - given that the order of sub-nodes does not matter, which is our case I reckon.